generated/nidaqmx/_grpc_interpreter.py

```diff
@@ -0,0 +1,1532 @@
+            self._client.CreateCOPulseChanTicks,
+            grpc_types.CreateCOPulseChanTicksRequest(self, task, counter, source_terminal, name_to_assign_to_channel, idle_state, initial_delay, low_ticks, high_ticks),
+        )
+
+    def create_co_pulse_chan_time(
+            self, task, counter, name_to_assign_to_channel, units, idle_state,
+            initial_delay, low_time, high_time):
+        response = self._invoke(
+            self._client.CreateCOPulseChanTime,
+            grpc_types.CreateCOPulseChanTimeRequest(self, task, counter, name_to_assign_to_channel, units, idle_state, initial_delay, low_time, high_time),
+        )
+
+    def create_di_chan(
+            self, task, lines, name_to_assign_to_lines, line_grouping):
+        response = self._invoke(
+            self._client.CreateDIChan,
+            grpc_types.CreateDIChanRequest(self, task, lines, name_to_assign_to_lines, line_grouping),
+        )
+
+    def create_do_chan(
+            self, task, lines, name_to_assign_to_lines, line_grouping):
+        response = self._invoke(
+            self._client.CreateDOChan,
+            grpc_types.CreateDOChanRequest(self, task, lines, name_to_assign_to_lines, line_grouping),
+        )
+
+    def create_lin_scale(
+            self, name, slope, y_intercept, pre_scaled_units, scaled_units):
+        response = self._invoke(
+            self._client.CreateLinScale,
+            grpc_types.CreateLinScaleRequest(self, name, slope, y_intercept, pre_scaled_units, scaled_units),
+        )
+
+    def create_map_scale(
+            self, name, prescaled_min, prescaled_max, scaled_min, scaled_max,
+            pre_scaled_units, scaled_units):
+        response = self._invoke(
+            self._client.CreateMapScale,
+            grpc_types.CreateMapScaleRequest(self, name, prescaled_min, prescaled_max, scaled_min, scaled_max, pre_scaled_units, scaled_units),
+        )
+
+    def create_polynomial_scale(
+            self, name, forward_coeffs, num_forward_coeffs_in, reverse_coeffs,
+            num_reverse_coeffs_in, pre_scaled_units, scaled_units):
+        response = self._invoke(
+            self._client.CreatePolynomialScale,
+            grpc_types.CreatePolynomialScaleRequest(self, name, forward_coeffs, num_forward_coeffs_in, reverse_coeffs, num_reverse_coeffs_in, pre_scaled_units, scaled_units),
+        )
+
+    def create_table_scale(
+            self, name, prescaled_vals, num_prescaled_vals_in, scaled_vals,
+            num_scaled_vals_in, pre_scaled_units, scaled_units):
+        response = self._invoke(
+            self._client.CreateTableScale,
+            grpc_types.CreateTableScaleRequest(self, name, prescaled_vals, num_prescaled_vals_in, scaled_vals, num_scaled_vals_in, pre_scaled_units, scaled_units),
+        )
+
+    def create_task(self, task, session_name):
+        response = self._invoke(
+            self._client.CreateTask,
+            grpc_types.CreateTaskRequest(self, task, session_name),
```

Comment:
Problems: 
- Protobuf message constructors require keyword arguments.
- Protobuf message constructors don't take a `GrpcStubInterpreter` (`self`).
- Request messages don't take out-parameters such as `task`.

```
Traceback (most recent call last):
  File "D:\dev\nidaqmx-python\examples\ai_voltage_sw_timed.py", line 15, in <module>
    task = interpreter.create_task(None, "MyTask")
  File "D:\dev\nidaqmx-python\generated\nidaqmx\_grpc_interpreter.py", line 843, in create_task
    grpc_types.CreateTaskRequest(self, task, session_name),
TypeError: No positional arguments allowed
```

I got my standalone script working by making these updates:
```
    def create_task(self, task, session_name):
        response = self._invoke(
            self._client.CreateTask,
            grpc_types.CreateTaskRequest(session_name=session_name),
        )
        return response.task

    def clear_task(self, task):
        response = self._invoke(
            self._client.ClearTask,
            grpc_types.ClearTaskRequest(task=task),
        )

    def create_ai_voltage_chan(
            self, task, physical_channel, name_to_assign_to_channel,
            terminal_config, min_val, max_val, units, custom_scale_name):
        response = self._invoke(
            self._client.CreateAIVoltageChan,
            grpc_types.CreateAIVoltageChanRequest(task=task, physical_channel=physical_channel, name_to_assign_to_channel=name_to_assign_to_channel, terminal_config=terminal_config, min_val=min_val, max_val=max_val, units=units, custom_scale_name=custom_scale_name),
        )

    def read_analog_scalar_f64(self, task, timeout):
        response = self._invoke(
            self._client.ReadAnalogScalarF64,
            grpc_types.ReadAnalogScalarF64Request(task=task, timeout=timeout),
        )
        return response.value
```

Reply:
Re: out-parameters: please read https://github.com/ni/nidaqmx-python/pull/286#discussion_r1170611132

The interpreter interface for `read_analog_f64` will need to take `read_array` by reference, so that clients can pass in a preallocated numpy array. However, the gRPC message `ReadAnalogF64Request` does not have a `read_array` field; that is only on the `ReadAnalogF64Response`. The gRPC interpreter implementation of `read_analog_f64` will have to assign `read_array` to have the value of `response.read_array`, while being careful to modify the passed-in ndarray rather than rebind it to point to the `response.read_array` field. I think you might want something like `read_array[:] = response.read_array`, *not* `read_array = response.read_array`.

Reply:
I hope this is going to be the case for all `read` functions which reads an `array`.